runtime/doc/pi_health.txt

```diff
@@ -23,60 +23,131 @@ Commands				*health-commands*
+		filetype-specific behavior, and standard plugins (including
+		:checkhealth).  If the runtime files cannot be found then
+		those features will not work.
 
 :checkhealth {plugins}
-                      Run healthcheck(s) for one or more plugins. E.g. to run
-                      only the standard Nvim healthcheck: >
-                        :checkhealth nvim
-<                      To run the healthchecks for the "foo" and "bar" plugins
-                      (assuming these plugins are on your 'runtimepath' and
-                      they have implemented health#foo#check() and
-                      health#bar#check(), respectively): >
-                        :checkhealth foo bar
+		Run healthcheck(s) for one or more plugins. E.g. to run only
+		the standard Nvim healthcheck: >
+			:checkhealth nvim
 <
+		To run the healthchecks for the "foo" and "bar" plugins
+		(assuming these plugins are on 'runtimepath' or 'packpath' and
+		they have implemented the Lua or Vimscript interface
+		require("foo.health").check() and health#bar#check(),
+		respectively): >
+			:checkhealth foo bar
+<
+		To run healthchecks for lua submodules, use dot notation or
+		"*" to refer to all submodules. For example nvim provides
+		`vim.lsp` and `vim.treesitter`  >
+			:checkhealth vim.lsp vim.treesitter
+			:checkhealth vim*
+<
+==============================================================================
+Lua Functions				*health-functions-lua* *health-lua*
+
+The Lua "health" module can be used to create new healthchecks (see also
+|health-functions-vim|). To get started, simply use:  >
+	local health = require('health')
+<
+health.report_start({name})				*health.report_start()*
+	Starts a new report. Most plugins should call this only once, but if
+	you want different sections to appear in your report, call this once
+	per section.
+
+health.report_info({msg})				*health.report_info()*
+	Reports an informational message.
+
+health.report_ok({msg})					*health.report_ok()*
+	Reports a "success" message.
+
+health.report_warn({msg} [, {advice}])			*health.report_warn()*
+	Reports a warning. {advice} is an optional List of suggestions.
+
+health.report_error({msg} [, {advice}])			*health.report_error()*
+	Reports an error. {advice} is an optional List of suggestions.
+
+==============================================================================
+Create a Lua healthcheck		*health-dev-lua*
+
+Healthchecks are functions that check the user environment, configuration,
+etc. Nvim has built-in healthchecks in $VIMRUNTIME/autoload/health/.
+
+To add a new healthcheck for your own plugin, simply define a Lua module in
+your plugin that returns a table with a "check()" function. |:checkhealth|
+will automatically find and invoke this function.
+
+If your plugin is named "foo", then its healthcheck module should be a file in
+one of these locations on 'runtimepath' or 'packpath':
+	- `lua/foo/health/init.lua`
+	- `lua/foo/health.lua`
```

Comment:
do not over-use backticks. They make it less convenient to copy-paste the names. Also the highlighting draws attention to stuff that often is not very important. Why highlight these filenames?